libs/ui-components/src/components/Device/DeviceDetails/DeviceDetailsPage.tsx

```diff
@@ -73,26 +83,33 @@ const DeviceDetailsPage = ({ children, hideTerminal }: DeviceDetailsPageProps) =
         </Nav>
       }
       actions={
-        <DetailsPageActions>
-          <DropdownList>
-            {canEdit && (
+        isEnrolled ? (
+          <DetailsPageActions>
+            <DropdownList>
+              {canEdit && (
+                <DropdownItem
+                  onClick={() => navigate({ route: ROUTE.DEVICE_EDIT, postfix: deviceId })}
+                  {...editActionProps}
+                >
+                  {t('Edit device configurations')}
+                </DropdownItem>
+              )}
+              {canDelete && deleteAction}
               <DropdownItem
-                onClick={() => navigate({ route: ROUTE.DEVICE_EDIT, postfix: deviceId })}
-                {...editActionProps}
+                onClick={() => {
+                  setShowDebugInfo(!showDebugInfo);
+                }}
               >
-                {t('Edit device configurations')}
+                {t('View status debug information')}
               </DropdownItem>
-            )}
-            {canDelete && deleteAction}
-            <DropdownItem
-              onClick={() => {
-                setShowDebugInfo(!showDebugInfo);
-              }}
-            >
-              {t('View status debug information')}
-            </DropdownItem>
-          </DropdownList>
-        </DetailsPageActions>
+              {canDecommission && decommissionAction}
+            </DropdownList>
+          </DetailsPageActions>
+        ) : (
+          <Button aria-label={t('Delete device forever')} variant="danger">
+            {deleteAction}
+          </Button>
```

Comment:
```suggestion
          canDelete && <Button aria-label={t('Delete device forever')} variant="danger">
            {deleteAction}
          </Button>
```

Reply:
You are right. 

And also, we should not allow to delete enrolled devices directly, so I also need to remove the `deleteAction` for enrolled devices.